src/hooks/usePriceFee.ts

```diff
@@ -1,39 +1,153 @@
+      return gasPrice
+    } catch (err) {
+      throw new Error("Failed to get gas price")
+    }
+  }
+
+  const getGasLimit = async () => {
+    if (l2Token.symbol === ETH_SYMBOL) {
+      return await getGasLimitGeneric(GatewayType.ETH_GATEWAY)
+    } else if (l2Token.symbol === WETH_SYMBOL) {
+      return await getGasLimitGeneric(GatewayType.WETH_GATEWAY)
+    } else {
+      return await getGasLimitGeneric(GatewayType.STANDARD_ERC20_GATEWAY)
+    }
+  }
+
+  const messageDataGeneric = contractName => {
+    let finalizeDepositParams: any = []
+    let finalizeDepositMethod = "finalizeDepositERC20"
+
+    if (contractName === GatewayType.ETH_GATEWAY) {
+      finalizeDepositParams = [walletCurrentAddress, walletCurrentAddress, amount, "0x"]
+      finalizeDepositMethod = "finalizeDepositETH"
+    } else if (contractName === GatewayType.WETH_GATEWAY) {
+      finalizeDepositParams = [
+        (l1Token as ERC20Token).address,
+        (l2Token as ERC20Token).address,
+        walletCurrentAddress,
+        walletCurrentAddress,
+        amount,
+        "0x",
+      ]
+    } else {
+      finalizeDepositParams = [
+        (l1Token as ERC20Token).address,
+        (l2Token as ERC20Token).address,
+        walletCurrentAddress,
+        walletCurrentAddress,
+        amount,
+        AbiCoder.defaultAbiCoder().encode(
+          ["bool", "bytes"],
+          [
+            true,
+            AbiCoder.defaultAbiCoder().encode(
+              ["bytes", "bytes"],
+              ["0x", AbiCoder.defaultAbiCoder().encode(["string", "string", "uint8"], [l2Token.symbol, l2Token.name, l2Token.decimals])],
+            ),
+          ],
+        ),
+      ]
+    }
+
+    return { finalizeDepositParams, finalizeDepositMethod }
+  }
+
+  const getGasLimitGeneric = async contractName => {
+    const { provider } = networksAndSigners[CHAIN_ID.L2]
+
+    const gateway = getContract(contractName, provider)
+    const l2messenger = getContract("SCROLL_MESSENGER", provider)
+    const { finalizeDepositMethod, finalizeDepositParams } = messageDataGeneric(contractName)
+    const message = gateway.interface.encodeFunctionData(finalizeDepositMethod, finalizeDepositParams)
+
+    const calldata = l2messenger.interface.encodeFunctionData("relayMessage", [
+      requireEnv(`REACT_APP_L1_${contractName}_PROXY_ADDR`), // l1 gateway
+      requireEnv(`REACT_APP_L2_${contractName}_PROXY_ADDR`), // l2 gateway
+      contractName === GatewayType.STANDARD_ERC20_GATEWAY ? 0 : amount,
+      ethers.MaxUint256,
+      message,
+    ])
+
+    const gaslimit = await provider.estimateGas({
+      from: BigNumber.from(requireEnv("REACT_APP_L1_SCROLL_MESSENGER")).add(BigNumber.from(OFFSET)).mod(BigNumber.from(2).pow(160)).toHexString(),
+      to: requireEnv(L2Contracts.SCROLL_MESSENGER.env),
+      data: calldata,
+    })
+
+    return (BigInt(gaslimit) * BigInt(120)) / BigInt(100)
+  }
+
+  return { getPriceFee, getGasLimit, getGasPrice }
```

Comment:
it seems like the only function we need is `getPriceFee`. Is there a situation where it's necessary to use `getGasLimit` and `getGasPrice ` separately?